addons/website_sale/__init__.py

```diff
@@ -17,3 +17,19 @@ def uninstall_hook(cr, registry):
     multi_company_rules = pl_rule or env['ir.rule']
     multi_company_rules += pl_item_rule or env['ir.rule']
     multi_company_rules.write({'active': True})
+
+
+def post_init_hook(cr, registry):
+    ''' Need to add filters to previously created website
+    '''
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    for website in env['website'].search([]):
+        action = env.ref('website_sale.dynamic_snippet_products_action', raise_if_not_found=False)
+        if action:
+            env['website.snippet.filter'].create({
+                'action_server_id': action.id,
+                'field_names': 'display_name,description_sale,image_512,list_price',
+                'limit': 16,
+                'name': _('Products'),
+                'website_id': website.id,
+            })
```

Comment:
no I didn't but why push it to 15.0 if its irrelevant anyways ?

Reply:
This is not what I meant, sorry.

Here is the scenario I had in mind:
- user uses this version => there are new filter records compared to before the fix
- user drops the product snippet(s) into its pages
- user upgrades odoo to v15
=> do the snippets dropped in v14 still work after the upgrade ?

If it is not the case, we might have to write an upgrade script to convert to v14 dropped snippet into one that is compatible with v15.

Reply:
well... this fix doesn't change the behavior of product snippet filters. if before the fix the user installed eCommerce first and then created a second website it will have the filter as normal. so I am not introducing new filter records, I am just adding them to all websites created before the installation. 

tldr:
this PR doesn't change the answer to your question. if snippets dropped in v14 work after the upgrade, they will still work after the upgrade, if not, then not.